### PR TITLE
quadratic sampling: separate diff from logits to filter out NaNs.

### PR DESCRIPTION
**Pull request by 50h100a**
Infs lead to NaNs in the logits which make samplers very sad.

### CODE REVIEW

**Review by AlpinDale — APPROVED** [review]
thanks for le fix
